### ISSUE TITLE
Template support

### ISSUE THREAD

**Issue by ultimate-tester** [thomasloven/hass-browser_mod#286]
Everyone agrees that browser_mod is absolutely great. I have used it today for the first time, and using some other plugins great things can be achieved. So thanks for that.

The only thing I find is that it clutters my yaml file extremely fast. I am currently only using it for light-popup-card that relies on the "popup_cards" key. For every light, I have a unique popup_cards entry with a lot of redundant configuration.

What I am proposing is templates for any key, but in my use case for popup_cards which is best explained in an example.

**My current code (per entity)**

```
popup_cards:
      light.living_room_standing_lamp:
        hide_header: true
        style:
          $: |
            .mdc-dialog .mdc-dialog__container {
              width: 100%;
            }
            .mdc-dialog .mdc-dialog__container .mdc-dialog__surface {
              width:100%;
              box-shadow:none;
            }
          .: |
            :host {
              --mdc-theme-surface: rgba(0,0,0,0);
              --secondary-background-color: rgba(0,0,0,0);
              --ha-card-background: rgba(0,0,0,0);
              --mdc-dialog-scrim-color: rgba(0,0,0,0.8);
              --mdc-dialog-min-height: 100%;
              --mdc-dialog-min-width: 100%;
              --mdc-dialog-max-width: 100%;
            }
            mwc-icon-button {
              color: #FFF;
            }
        card:
          type: custom:light-popup-card
          entity: light.living_room_standing_lamp
```

**Templated code (can be defined at the top level of the yaml config)**

```
popup_cards_templates:
      my_popup_card_template:
        hide_header: true
        style:
          $: |
            .mdc-dialog .mdc-dialog__container {
              width: 100%;
            }
            .mdc-dialog .mdc-dialog__container .mdc-dialog__surface {
              width:100%;
              box-shadow:none;
            }
          .: |
            :host {
              --mdc-theme-surface: rgba(0,0,0,0);
              --secondary-background-color: rgba(0,0,0,0);
              --ha-card-background: rgba(0,0,0,0);
              --mdc-dialog-scrim-color: rgba(0,0,0,0.8);
              --mdc-dialog-min-height: 100%;
              --mdc-dialog-min-width: 100%;
              --mdc-dialog-max-width: 100%;
            }
            mwc-icon-button {
              color: #FFF;
            }
        card:
          type: custom:light-popup-card
```

**Usage (in the entity)**

```
popup_cards:
   light.living_room_standing_lamp: my_popup_card_template
   card:
      entity: light.living_room_standing_lamp
```

This reduces the code per-entity and therefore the total yaml size drastically.

**Comment by jherby2k** [comment]
This would be awesome! Any ideas on how to do this maybe using config-template-card?

**Comment by thomasloven** [comment]
Closed because of Browser Mod 2.0 release.
Please see #354.
